### ISSUE TITLE
Problem in starting Refprop 10.0

### ISSUE THREAD

**Issue by XiuzhongShen** [usnistgov/REFPROP-wrappers#349]
I am a user of REFPROP version 9.0. I bought REFPROP UPGRADE version 10 one year ago. I followed the installing process described in See the REFPROP FAQ for help installing and using REFPROP. It looks that I have finished installing REFPROP UPGRADE version 10. But I cannot use the Refprop version 10 in Excel and Matlab R2019b, even in the original REFPROP.XLS. My operating system and Version are Windows 10 Pro Version 1909. In order to solve this problem, recently I bought REFPROP version 10. I uninstalled  the REFPROP UPGRADE version 10 and installed the REFPROP version 10. However, the problem still cannot be solved. I hope that a kind people can tell me a way to solve this problem?

**Comment by ianhbell** [comment]
Can you please fill out the template?  We ask people to fill out out so that we have the necessary information to help you.  

What is not working, and what error messages do you get?

**Comment by XiuzhongShen** [comment]
In sheet of Available Properties in REFPROP.XLS, "#VALUE!"( value error) appears in all cells that need to be calculated. The values are not shown correctly in that sheet.

**Comment by XiuzhongShen** [comment]
Please see the following photo of the sheet.
![REFPROP](https://user-images.githubusercontent.com/78000396/105795725-a621c780-5fd0-11eb-9799-236064e784ed.png)

**Comment by ianhbell** [comment]
Please start here: https://github.com/usnistgov/REFPROP-wrappers/issues/105

**Comment by XiuzhongShen** [comment]
Thank you very much for your comment. I have spent 2 days to study #105 and check Items 1-16 in Section 32 of "Resolving Problems Linking with Excel" and Section 33 of "Other Issues Related to the Use of Refprop in Excel"  in "Answers to Frequently Asked Questions". I have made a lot tries to solve this problem. But the problem still persists now and it may be difficult for me solve this problem independently. Would you please log into my machine remotely to find the problem and the solution to the problem for me?

**Comment by EricLemmon** [comment]
OK, I'll contact you directly.

**Comment by NickAtNight62** [comment]
Well, you have the expert (Eric) involved now.  

I can't read the Chinese on the menu.  

There should be a "REFPROP"  tab added to the MENU.  It does not seem to be there

Is the Menu addition on the screen?   

![Screenshot 2021-01-28 093055](https://user-images.githubusercontent.com/58614082/106160611-a1cde980-614b-11eb-96a8-8e71877e3363.png)

**Comment by NickAtNight62** [comment]
From the "Developer" tab
- Select "Excel Add-ins"
- Verify that the two REFPROP Addins are installed
---   "REFPROP AddIns"   (this runs the calculations)
--- "REFPROP Ribbon"  (this runs the Menu and the Wizard)

![Screenshot 2021-01-28 093821](https://user-images.githubusercontent.com/58614082/106161469-a6df6880-614c-11eb-9ecb-2911e0f05a83.png)

**Comment by NickAtNight62** [comment]
The add-ins necessary to run the calculations are located in:   
      C:\Program Files (x86)\REFPROP

The file names are:
  REFPROP.XLA is the REFPROP AddIns
  PREFPROP_Ribbon.xlam is the REFPROP Ribbon

If the Ribbon is running, you should be able to run the "Wizard" (see picture)

![Screenshot 2021-01-28 104734](https://user-images.githubusercontent.com/58614082/106170678-4ce3a080-6156-11eb-8940-e49630823fe9.png)

With the default Wizard, Click the calculate button.
- If the AddIn is working, it should calculate the density of Air (1.276 kg/m3)

If that works, try the Compose button.  
- That will place a properly formulated calculation into the spreadsheet. It should like:
                  "Density  1.27615426  kg/m^3"
- The properly formulate calculation is:
        =refprop("D","Air","TP","SI",273.15,0.1,0)
- You should be able to cut and paste and put that equation into your spreadsheet manually.  If the program is working, then you should get a result.

**Comment by XiuzhongShen** [comment]
Thank you very much for your help. There is no "Developer" and "REFPROP" in my EXCEL menu. I cannot find the Refprop AddIns through "file"->"Option"->"Addins". There is no "PI Wizard" near "Insert function" in the equation menu.

**Comment by XiuzhongShen** [comment]
Both of REFPROP.XLA and PREFPROP_Ribbon.xlam files exist in the fold of C:\Program Files (x86)\REFPROP. It looks that they do not work there since REFPROP Wizard does not appear in the EXCEL.

**Comment by XiuzhongShen** [comment]
![2021-01-29](https://user-images.githubusercontent.com/78000396/106217035-b7104a00-6217-11eb-8f40-5f1e366e43f2.png)

**Comment by ianhbell** [comment]
When go to the Addins panel, and Browse, can you browse to find the addin in the REFPROP folder?

**Comment by XiuzhongShen** [comment]
I have found the REFPROP AddIns in the REFPROP folder and added it just now. But the Excel file still does not work.

**Comment by XiuzhongShen** [comment]
I have added the excel AddIns of REFPROP AddIns and REFPROP Ribbon now. But the errors of #VALUE! do not disappear in the REFPROP.XLS.

**Comment by NickAtNight62** [comment]
The "Developer" tab is not shown by default.  

To turn it on:    
   File
     Options
        Customize Ribbon
             Check the box for the developer tab.

**Comment by NickAtNight62** [comment]
It has been a long while since I installed REFPROP, but I believe that the two addins were added automatically when the program was installed.  Perhaps Eric can refresh my memory.

Addins can be installed but not active.  Make sure that the two files are checked.

There may be a security program disabling the addins and preventing them from running. 
- If you have computer IT/security team, ask them first.  There are many ways to disable macro's from running. so it would be best to ask the inhouse experts first, as they would have likely been the ones to setup the block.

If not, you could start by looking in the 'Macro Security' section

For example, in "Trusted Locations"  - I have the User Location  C:\Program Files (x86)\REFPROP\ on my list.
Note that there is also a check box at the bottom that allows me to 'Disable all Trusted location'

![Screenshot 2021-01-28 211834](https://user-images.githubusercontent.com/58614082/106227263-8811be00-61ae-11eb-89b5-4fd1eef91403.png)

So if I turn some of those features off and restart...   Well, that gives me a #name problem.  
- Not quite the same problem as you have, you have a #Value! problem.

![Screenshot 2021-01-28 212504](https://user-images.githubusercontent.com/58614082/106227761-6cf37e00-61af-11eb-91c2-e64012ddf8d9.png)

**Comment by XiuzhongShen** [comment]
Thank you (Ian and Nick) very much. Eric has helped me to turn on the "REFPROP" tab remotely. Following the comment of Nick, I have turned on the "Developer" tab. But in the REFPROP Wizard, I still cannot obtain the air density of 1.276 kg/m3). The error message is "#Make sure the REFPROP.XLA add-in is enabled to calculate items in the cells." Actually, I have enabled REFPROP add-in. Because it was too late today, Eric will help me to solve the remaining problem tomorrow. I really appreciate your kind help.

![2021-01-29 (4)](https://user-images.githubusercontent.com/78000396/106261990-ac7ca180-6265-11eb-9304-be0f4c9c77c7.png)

**Comment by NickAtNight62** [comment]
Excellent.  So one of the addin's is now working.  So now we are just down to an issue with the refprop.xla addin.

**Comment by NickAtNight62** [comment]
A side task.    Write your first VBA Macro - {**V**isual **B**asic for **A**pplications}

1. Developer tab
2. Record Macro
3. Complete the Record Macro Box.
- Enter your Macro name:   "MyFirstMacro"
- Enter your Shortcut key:   Ctrl+  "q"
- Store macro in:  "This Workbook"
- Description:   "This is my first macro.  I made it by Recording a sequence of steps using the Record Macro button"

Perform the actions you want to record.
Press "Stop Recording"

For this example, just enter the numbers 1 to 10... one number on each row.

1
2
3
4
5
6
7
8
9
10

To run your Macro, press Ctrl + q

Erase the spreadsheet.   Run the Macro again.

Erase the spreadsheet.  Press the Macro button.  That will show you a  list of all of your Macros.   "MyFirstMacro" should be the only one on the list.

Select the Macro you want to run from the list, Press the RUN button to run from this list.

If you want to add it to your ribbon...

Right click on any ribbon - Customize Ribbon
- Choose "Macros"  Find "MyFirstMacro" - add it to the desired ribbon.
- Notice I have a Custom tab called "KLEA" and a Custom Group called "TECH"

![Screenshot 2021-01-29 143048](https://user-images.githubusercontent.com/58614082/106324445-c35bcd80-623e-11eb-9793-ab62b2a7b404.png)

Now you have a button you can press to run the Macro.

Go to Developer
Select Visual Basic
Find your workbook (VBAProject (Book1))
   Modules
      Module 1

Here is the code you wrote:

Sub MyFirstMacro()
'
' MyFirstMacro Macro
' This is my first macro.  I made it by Recording a sequence of steps using the Record Macro button
'
' Keyboard Shortcut: Ctrl+q
'
    ActiveCell.FormulaR1C1 = "1"
    Range("B4").Select
    ActiveCell.FormulaR1C1 = "2"
    Range("B5").Select
    ActiveCell.FormulaR1C1 = "3"
    Range("B6").Select
    ActiveCell.FormulaR1C1 = "4"
    Range("B7").Select
    ActiveCell.FormulaR1C1 = "5"
    Range("B8").Select
    ActiveCell.FormulaR1C1 = "6"
    Range("B9").Select
    ActiveCell.FormulaR1C1 = "7"
    Range("B10").Select
    ActiveCell.FormulaR1C1 = "8"
    Range("B11").Select
    ActiveCell.FormulaR1C1 = "9"
    Range("B12").Select
    ActiveCell.FormulaR1C1 = "10"
    Range("B13").Select
End Sub

Now you can manually edit it to make it do something different, or fix any mistakes you made.

**Comment by XiuzhongShen** [comment]
Thank you (Nick) very much. My problem has been solved completely by following your comment. My REFPROP.XLS can work properly now as follows.  I have practiced writing my first VBA Macro by following your side task. I have successfully finished the side task. I really appreciate the time and effort of your 3 experts (Eric, Ian and Nick) in helping fixing my problem.
 
![2021-01-30 (2)](https://user-images.githubusercontent.com/78000396/106344646-60297400-62ee-11eb-96de-5d5b2a2544ec.png)
